src/PrestaShopBundle/Form/Admin/Type/EventListener/DisablingToggleListener.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type\EventListener;
+
+use PrestaShopBundle\Form\Admin\Extension\DisablingToggleExtension;
+use PrestaShopBundle\Form\Admin\Type\DisablingToggleType;
+use PrestaShopBundle\Form\FormCloner;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+class DisablingToggleListener implements EventSubscriberInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            FormEvents::PRE_SET_DATA => 'addDisablingToggle',
+            FormEvents::PRE_SUBMIT => 'addDisablingToggle',
+        ];
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function addDisablingToggle(FormEvent $event): void
+    {
+        $form = $event->getForm();
+        $parent = $form->getParent();
+        $fieldName = DisablingToggleExtension::FIELD_PREFIX . $form->getName();
+        if ($parent->has($fieldName)) {
+            return;
+        }
+
+        $parent->add($fieldName, DisablingToggleType::class);
+        $shouldBeDisabled = !$parent->get($fieldName)->getData();
+
+        $formCloner = new FormCloner();
+        $newOptions = [
+            'disabled' => $shouldBeDisabled,
+            'attr' => [
+                'data-toggled-by' => $fieldName,
+            ],
+        ];
+
+        foreach ($form->all() as $childForm) {
```

Comment:
This should be placed into a method, and the method should be called recursively on `$childForm` because with this implementation the disabled property is only propagated to the first layer of children not all recursive children